packages/dendron-plugin-views/src/components/DendronConfigure.tsx

```diff
@@ -0,0 +1,175 @@
+
+  useEffect(() => {
+    setConfig(engine.config);
+    Object.keys(dendronConfig).forEach(
+      (key) => (dendronConfig[key].default = _.get(engine.config, key))
+    );
+  }, [engine.config]);
+
+  useConfig({ opts: workspace });
+  const { Header, Content, Sider } = Layout;
+  const items = Object.keys(dendronConfig).map((conf) => {
+    return {
+      key: conf,
+      label: _.startCase(dendronConfig[conf].group),
+    };
+  });
+  const menuItems = items.filter(
+    (value, index, self) =>
+      index === self.findIndex((t) => t.label === value.label)
+  );
+
+  const cleanTitle = (title: string) => {
+    return title
+      .split(".")
+      .map((k) => _.startCase(k))
+      .join(" > ");
+  };
+
+  const postMessage = ({ key, value }: { key: string; value: any }) => {
+    let newConf = _.cloneDeep(config);
+    newConf = newConf && _.set(newConf, key, value);
+    dendronConfig[key].default = value;
+    setConfig(newConf);
+    postVSCodeMessage({
+      type: ConfigureUIMessageEnum.onUpdateConfig,
+      data: { config: newConf },
+      source: DMessageSource.webClient,
+    } as ConfigureUIMessage);
+  };
+
+  const handleSearch = debounce((e: any) => {
+    setSearchString(e.target.value);
+  }, 500);
+
+  return (
+    <Layout style={{ height: "100vh", overflow: "hidden" }}>
+      {!config ? (
+        <Spin />
+      ) : (
+        <>
+          <Header className="header">
+            <Input placeholder="search config" onChange={handleSearch} />
+          </Header>
+          <Layout>
+            <Sider
+              width={160}
+              style={{
+                height: "100%",
+                borderRight: "1px solid #383838",
+                margin: 0,
+                padding: "20px 16px 0 21px",
+              }}
+            >
+              <Anchor
+                getContainer={() =>
+                  document.getElementById("configure-content")!
+                }
+              >
+                <Space
+                  direction="vertical"
+                  size="middle"
+                  style={{ display: "flex" }}
+                >
+                  {menuItems.map((item) => (
+                    <Anchor.Link
+                      key={item.key}
+                      href={`#${item.label.toLowerCase()}`}
+                      title={item.label}
+                    />
+                  ))}
+                </Space>
+              </Anchor>
+            </Sider>
+            <Layout>
+              <Content
+                id="configure-content"
+                style={{
+                  padding: 24,
+                  margin: 0,
+                  minHeight: 280,
+                  height: "100%",
+                  overflowY: "scroll",
+                }}
+              >
+                <Space
+                  direction="vertical"
+                  size="middle"
+                  style={{ display: "flex" }}
+                >
+                  {Object.keys(dendronConfig)
```

Comment:
can we have two levels of grouping if it exists? (eg. `commands.lookup.note` should be in one box) https://www.loom.com/i/ffc413e6c6f04cd3af9d1af329aa0ebf

Reply:
@kevin, can we do this in a follow-up PR? might need to add more than just a tweak to get this work.